phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2112,6 +2116,17 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
             if (tableType != PTableType.INDEX && updateCacheFrequencyProp != null) {
                 updateCacheFrequency = updateCacheFrequencyProp;
             }
+
+            Integer viewTTL = null;
+            Long viewTTLHighWaterMark = null;
+            Integer viewTTLProp = null;
+
+            //TODO Look at the VIEW_TTL table property values that are set instead of hardcoded defaults.
+            if (!SchemaUtil.isSystemTable(Bytes.toBytes(SchemaUtil.getTableName(schemaName, tableName))) && viewTTLProp != null) {
+                viewTTL = HConstants.FOREVER;
+                viewTTLHighWaterMark = 0L;
+            }
```

Comment:
Also, we would need some sanity checks for the passed in viewTTL and viewTTLHWM properties, for example: cannot be negative, some possible max value, etc.